index.bs

```diff
@@ -11876,33 +11892,24 @@ the [=primary interface=]
 of the platform object.
 
 
-<h4 id="indexed-and-named-properties">Indexed and named properties</h4>
+<h3 id="es-legacy-platform-objects" oldids="indexed-and-named-properties">Legacy platform objects</h3>
 
-If a [=platform object=] implements an [=interface=] that
-[=support indexed properties|supports indexed=] or
-[=support named properties|named properties=],
-the object will appear to have additional properties that correspond to the
-object’s indexed and named properties.  These properties are not “real” own
+[=Legacy platform objects=] will appear to have additional properties that correspond to the
+[=indexed properties|indexed=] and [=named properties=].  These properties are not “real” own
 properties on the object, but are made to look like they are by being exposed
 by the \[[GetOwnProperty]] internal method.
 
-However, when the [{{Global}}] or
-[{{PrimaryGlobal}}]
-[=extended attribute=] has been used,
-named properties are not exposed on the object but on another object
-in the prototype chain, the [=named properties object=].
-
 It is permissible for an object to implement multiple interfaces that support indexed properties.
```

Comment:
You mean "more than one interface that supports indexed/named properties"?

Gecko's IDL parser already disallows this (in violation of the above-quoted spec text) unless the interfaces inherit from each other.  There _is_ one case of interfaces inheriting from each other that both support named properties: HTMLFormControlsCollection and HTMLCollection.  There are no such cases for indexed properties.

In practice, I think the sane thing to do would be to disallow getters/setters on mixins altogether.  Then we only have to worry about the inheritance case, in which behavior is at least well-defined: the most-derived thing wins.  That said, we could disallow shadowing of indexed getters/setters too, if we want to, since nothing does that right now and adding new things that do it would be ... undesirable.